File: Students/John/python/labs/lab02.py
"""

all codes are written and created by John Robson Wed Mar 10, 2021

"""

import random

while True:
    noun_list = input("List 3 nouns: ").split()
    adj_list = input("List 3 emotions: ").split()
    clothing_list = input("list 3 articles of clothing: ").split()
    
    noun = random.choice(noun_list)
    adjective = random.choice(adj_list)
    clothing = random.choice(clothing_list)

    lyrics = f"""
    Be our {noun}
    Be our {noun}
    Turning 60 is {adjective}
    Tighten up your bow tie round your neck and wear your favorite {clothing}
    """
    print(lyrics);
    play_again = input("Would you like to continue? ")
    if play_again.lower() == "no":
        break

File: Students/John/javascript/python-labs/lab02.js
/*

all codes are written and created by John Robson Thu Mar 18, 2021

*/

export function runLab02() {
  renApp(
    [
      { el: "h1", text: "Lab 02 - Mad Libs" },
      { el: "label", text: getText("nouns") },
      { el: "input", cl: "noun" },
      { el: "label", text: getText("emotions") },
      { el: "input", cl: "adj" },
      { el: "label", text: getText("articles of clothing") },
      { el: "input", cl: "clothing" },
      { el: "button", text: "Submit", cl: "grade-btn" },
    ],
    $("main")
  );

  $(".grade-btn").onclick = getValues;
}

function getValues() {
  let validation;

  $$("input").forEach((input) => {
    if (input.value === "") validation = false;
  });

  if (!validation)
    renApp(
      { el: "p", cl: "error", text: "Please fill out all fields." },
      $("main")
    );

  const nounsList = $(".noun").value.split(","),
    adjList = $(".adj").value.split(","),
    clothingList = $(".clothing").value.split(",");

  printLyrics(nounsList, adjList, clothingList);
}

function printLyrics(nouns, adjs, clothes) {
  const h2 = ren({ el: "h2", text: "Results: " }),
    pText = `Be our ${rand(nouns)}
    Be our ${rand(nouns)}
    Turning 60 is ${rand(adjs)}
    tighten up your bow tie around your neck and wear your favorite ${rand(
      clothes
    )}`,
    p = ren({ el: "p", text: pText }),
    p2 = ren({ el: "p", text: "Refresh page to play again." });

  clear($("main"));
  app([h2, p, p2], $("main"));
}

function getText(word) {
  return "Please enter some " + word + " separated by a comma";
}
